crates/holochain/src/core/ribosome/host_fn/emit_signal.rs

```diff
@@ -1,14 +1,20 @@
-use crate::core::ribosome::error::RibosomeResult;
 use crate::core::ribosome::CallContext;
 use crate::core::ribosome::RibosomeT;
+use crate::core::{ribosome::error::RibosomeResult, signal::Signal};
 use holochain_zome_types::EmitSignalInput;
 use holochain_zome_types::EmitSignalOutput;
 use std::sync::Arc;
 
 pub fn emit_signal(
     _ribosome: Arc<impl RibosomeT>,
-    _call_context: Arc<CallContext>,
-    _input: EmitSignalInput,
+    call_context: Arc<CallContext>,
+    input: EmitSignalInput,
 ) -> RibosomeResult<EmitSignalOutput> {
-    unimplemented!();
+    tokio_safe_block_on::tokio_safe_block_forever_on(async move {
```

Comment:
This can be removed once the async on `send(` is removed. See my comment above.

Reply:
Good catch